### PR TITLE
add ledger testcase

### PR DESCRIPTION
**Pull request by lucas7788**

### CODE REVIEW

File: sharding/testcases/ledger/ledger-testcase.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
+# 账本测试用例
+
+## 账本介绍
+Shard中会执行root区块和shard区块，因此有root和shard两个账本。
+为什么需要root账本？
+
+* 每个shard可以有不同version／height的root账本，保证分片中的状态一致性
```
Comment (qiluge) [inline sharding/testcases/ledger/ledger-testcase.md]: what's this means?

File: sharding/testcases/ledger/ledger-testcase.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+## 账本介绍
+Shard中会执行root区块和shard区块，因此有root和shard两个账本。
+为什么需要root账本？
+
+* 每个shard可以有不同version／height的root账本，保证分片中的状态一致性
+* 实现shard账本与root block height状态的绑定
+
+## 测试用例
+
+#### case1, shard信息保存在rootshard上
```
Comment (qiluge) [inline sharding/testcases/ledger/ledger-testcase.md]: please describe the detail. What info? And what is destination of this tx?

File: sharding/testcases/ledger/ledger-testcase.md [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+#### 步骤
+
+1. 创建好shard, 激活该shard
+2. 启动shard， 记录下启动shard时对应的rootshard 区块高度
+3. 查看账本的变化(Chain文件的变化)
+
+#### 预期
+
+shard会先启动同步线程，同事生成rootshard Chain文件，此时shard并不会出块，也就不会有shard账本，待同步到启动shard的区块高度时，shard开始出块，shard ledger开始生成
```
Comment (qiluge) [inline sharding/testcases/ledger/ledger-testcase.md]: "同事" -> 同时

File: sharding/testcases/ledger/ledger-testcase.md (hunk 1 above)
Comment (lucas7788) [reply sharding/testcases/ledger/ledger-testcase.md]: For details, please refer to shard design doc, ledger design

Comment (xiemylogos) [reply sharding/testcases/ledger/ledger-testcase.md]: suggest change root name into parent.like there are parent ledger and shard ledger

File: sharding/testcases/ledger/ledger-testcase.md [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+rootshard推出来的shard治理事件消息保存在rootshard的eventstore中。
+该类事件有shard配置更新事件、shard激活事件。
+#### 步骤
+1. 构造shard config更新的交易和shard激活交易
+2. 将交易发往rootshard，等待落账
+3. 根据shardId和rootshard的高度,查询event, 通过rootshard查询
+
+#### 预期
+
+通过rootshard查询成功
```
Comment (xiemylogos) [inline sharding/testcases/ledger/ledger-testcase.md]: modify rootshard to  name rootchain?

File: sharding/testcases/ledger/ledger-testcase.md [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+2. 启动shard， 记录下启动shard时对应的rootshard 区块高度
+3. 查看账本的变化(Chain文件的变化)
+
+#### 预期
+
+shard会先启动同步线程，同时生成rootshard Chain文件，此时shard并不会出块，也就不会有shard账本，待同步到启动shard的区块高度时，shard开始出块，shard ledger开始生成
+
+
+### case3, rootshard推出来的子shard事件消息(分片治理相关的事件)
+rootshard推出来的shard治理事件消息保存在rootshard的eventstore中。
```
Comment (xiemylogos) [inline sharding/testcases/ledger/ledger-testcase.md]: shard chain sync root chain block info,save some shard gov event to shard parent ledger?

**Comment by qiluge** [comment]
rebase error, close it

**Pull request closed without merging**